solution-base/build.sh

```diff
@@ -0,0 +1,205 @@
+set -eu
+
+PWD=$(pwd)
+BUILD_ROOT=${PWD}/_build
+ISO_ROOT=${BUILD_ROOT}/root
+IMAGES_ROOT=${ISO_ROOT}/images
+DEPLOY_ROOT=${ISO_ROOT}/operator/deploy
+
+PRODUCT_NAME=Zenko-Base
+PRODUCT_LOWERNAME=zenko-base
+BUILD_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+BUILD_HOST=$(hostname)
+
+VERSION_SHORT=$(git describe --abbrev=0)
+VERSION_FULL=${VERSION_SHORT}-dev
+GIT_REVISION=$(git describe --long --always --tags --dirty)
+ISO=${BUILD_ROOT}/${PRODUCT_LOWERNAME}-${VERSION_FULL}.iso
+
+DOCKER=docker
+DOCKER_OPTS=
+DOCKER_SOCKET=${DOCKER_SOCKET:-unix:///var/run/docker.sock}
+HARDLINK=hardlink
+SKOPEO=skopeo
+SKOPEO_OPTS="--override-os linux --insecure-policy"
+OPERATOR_TAG=$(grep /operator: deps.txt | awk -F ':' '{print $2}')
+SOLUTION_REGISTRY=metalk8s-registry-from-config.invalid/${PRODUCT_LOWERNAME}-${VERSION_FULL}
+
+ONESSL_VERSION=v0.13.1
+
+export KUBEDB_SCRIPT_LOCATION="curl -fsSL https://raw.githubusercontent.com/kubedb/installer/v0.13.0-rc.0/"
+export KUBEDB_NAMESPACE=kube-system
+export KUBEDB_SERVICE_ACCOUNT=kubedb-operator
+export KUBEDB_OPERATOR_NAME=operator
+export KUBEDB_ENABLE_RBAC=true
+export KUBEDB_RUN_ON_MASTER=0
+export KUBEDB_ENABLE_VALIDATING_WEBHOOK=false
+export KUBEDB_ENABLE_MUTATING_WEBHOOK=false
+export KUBEDB_DOCKER_REGISTRY=${SOLUTION_REGISTRY}
+export KUBEDB_OPERATOR_TAG=v0.13.0-rc.0
+export KUBEDB_IMAGE_PULL_POLICY=IfNotPresent
+export KUBEDB_ENABLE_ANALYTICS=false
+export KUBEDB_ENABLE_STATUS_SUBRESOURCE=false
+export KUBEDB_BYPASS_VALIDATING_WEBHOOK_XRAY=false
+export KUBEDB_USE_KUBEAPISERVER_FQDN_FOR_AKS=true
+export KUBEDB_PRIORITY_CLASS=system-cluster-critical
+
+# grab our dependencies from our deps.txt file as an array
+readarray -t DEP_IMAGES < deps.txt
+
+function clean()
+{
+    echo cleaning
+    rm -rf ${BUILD_ROOT}
+    rm -rf ca.crt ca.key server.crt server.key
+}
+
+function mkdirs()
+{
+    echo making dirs
+    mkdir -p ${ISO_ROOT}
+    mkdir -p ${IMAGES_ROOT}
+    mkdir -p ${DEPLOY_ROOT}
+}
+
+onessl_found() {
+    echo checking onessl
+    # https://stackoverflow.com/a/677212/244009
+    if [ -x "$(command -v onessl)" ]; then
+        onessl version --check=">=${ONESSL_VERSION}" >/dev/null 2>&1 || {
+            # old version of onessl found
+            echo "Found outdated onessl"
+            return 1
+        }
+        export ONESSL=onessl
+        return 0
+    fi
+    return 1
+}
+
+function gen_certs()
+{
+    echo generating certs
+    onessl create ca-cert
+    onessl create server-cert server --domains=kubedb-$KUBEDB_OPERATOR_NAME.$KUBEDB_NAMESPACE.svc
+    export SERVICE_SERVING_CERT_CA=$(cat ca.crt | onessl base64)
+    export TLS_SERVING_CERT=$(cat server.crt | onessl base64)
+    export TLS_SERVING_KEY=$(cat server.key | onessl base64)
+}
+
+function kubedb_yamls()
+{
+    ${KUBEDB_SCRIPT_LOCATION}deploy/operator.yaml | onessl envsubst > ${DEPLOY_ROOT}/operator.yaml
+    # ${KUBEDB_SCRIPT_LOCATION}deploy/kubedb-catalog/mongodb.yaml | onessl envsubst > deploy/crds/mongodb.yaml
```

Comment:
the kubedb catalog for mongo includes a lot of different docker images, I am not sure if I should include the entire catalog or not as we would need to also grab the images for all versions we pulled and it seemed like a lot. 
( here is the full file for reference https://raw.githubusercontent.com/kubedb/installer/v0.13.0-rc.0/deploy/kubedb-catalog/mongodb.yaml )

Reply:
it should be safe to apply the full catalog. As long as those unused versions aren't being pulled, it is fine to leave them in. (🤔 though it does create useless resources)

Reply:
Ok so use the full catalog but only include one set of images?

Reply:
^ yeah. I believe that will work=

Reply:
ok done, did some clean up as well